frontend/ui/uimanager.lua

```diff
@@ -1439,11 +1439,13 @@ end
 
 -- Process all pending events on all registered ZMQs.
 function UIManager:processZMQs()
-    if self._zeromqs[1] then
-        self.event_hook:execute("InputEvent")
-    end
+    local sent_InputEvent = false
     for _, zeromq in ipairs(self._zeromqs) do
         for input_event in zeromq.waitEvent, zeromq do
+            if not sent_InputEvent then
+                self.event_hook:execute("InputEvent")
+                sent_InputEvent = true
+            end
```

Comment:
Umm, this chunk doesn't make sense to me?

AFAICT, it does the exact same thing, but.... uglier? ^^.

Reply:
Is the intent to only send an `InputEvent` in case `waitEvent` actually returns something?

Because that raises an interesting conundrum: AFAICT, none of our (ZMQ) waitEvent implementations actually ever return anything... ^^.

Reply:
This may happen when the cat let the mice dance :)
Haven't reread what I wrote to explain the issue and this solution, but will if needed :) It's the 2nd bullet point in the first link, solution detailed in the 2nd link.
https://github.com/koreader/koreader/pull/10223#issuecomment-1928233249
https://github.com/koreader/koreader/pull/10223#issuecomment-1933622780

As you're back, there's still the Ctrl-C and the "not painting 1 covered widget" that you may have some thoughts about.

Reply:
> * The main use-case for these is for AutoSuspend's sake, to make it aware that we're actually active and do _not_ want to enter suspend/standby.

We might instead want to tackle this the other way around:

* Extend AutoSuspend to provide a way to stop/resume its timers (a pair of events?), and simply call that whenever it makes sense?

Reply:
We have UIManager:preventStandby() / UIManager:allowStandby() - isn't that what you wish for?

Reply:
That's for standby only, not suspend ;).

And, now that I think about it, standby is already skipped if wifi is enabled, which is almost guaranteed for these use-cases ;).

Reply:
Hm? A server can happily run locally on the device without any wifi.

Reply:
Hence my "*almost* guaranteed" comment ^^